dependency_cache_test.go

```diff
@@ -317,6 +319,21 @@ func testDependencyCache(t *testing.T, context spec.G, it spec.S) {
 			Expect(io.ReadAll(a)).To(Equal([]byte("alternate-fixture")))
 		})
 
+		it("sets downloaded file name to uri's sha256 with empty SHA256 and query parameters in the uri", func() {
```

Comment:
- [ ] add a new test for a normal uri without query params and with non empty SHA256